service/src/main/java/greencity/service/HabitStatusServiceImpl.java

```diff
@@ -29,60 +31,50 @@
 public class HabitStatusServiceImpl implements HabitStatusService {
     private final HabitStatusRepo habitStatusRepo;
     private final HabitStatusCalendarService habitStatusCalendarService;
+    private final HabitStatusCalendarRepo habitStatusCalendarRepo;
     private final ModelMapper modelMapper;
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public HabitStatusDto getById(Long habitAssignId) {
-        return modelMapper.map(habitStatusRepo.findById(habitAssignId)
-                .orElseThrow(() -> new WrongIdException(ErrorMessage.HABIT_ASSIGN_NOT_FOUND_BY_ID + habitAssignId)),
+    public HabitStatusDto getById(Long id) {
+        return modelMapper.map(habitStatusRepo.findById(id)
+                .orElseThrow(() -> new WrongIdException(ErrorMessage.NO_STATUS_FOR_SUCH_HABIT_ASSIGN)),
             HabitStatusDto.class);
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public void saveStatusByHabitAssign(HabitAssignDto habitAssign) {
+    public void saveStatusByHabitAssign(HabitAssignVO habitAssign) {
         HabitStatus habitStatus = HabitStatus.builder()
             .habitStreak(0)
             .habitAssign(modelMapper.map(habitAssign, HabitAssign.class))
             .workingDays(0)
             .lastEnrollmentDate(LocalDateTime.now())
             .build();
-
         habitStatusRepo.save(habitStatus);
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public HabitStatusDto findStatusByHabitAssignId(Long habitAssignId) {
-        return modelMapper.map(habitStatusRepo.findByHabitAssignId(habitAssignId)
-                .orElseThrow(() -> new WrongIdException(ErrorMessage.NO_STATUS_FOR_SUCH_HABIT_ASSIGN)),
-            HabitStatusDto.class);
-    }
-
     /**
      * {@inheritDoc}
      */
     @Override
     public HabitStatusDto findActiveStatusByUserIdAndHabitId(Long userId, Long habitId) {
         return modelMapper.map(habitStatusRepo.findByUserIdAndHabitId(userId, habitId)
-                .orElseThrow(() -> new WrongIdException(ErrorMessage.USER_HAS_NO_STATUS_FOR_SUCH_HABIT)),
+                .orElseThrow(() -> new WrongIdException(ErrorMessage.NO_STATUS_FOR_SUCH_HABIT_ASSIGN)),
             HabitStatusDto.class);
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public HabitStatusDto findStatusByUserIdAndHabitIdAndCreateDate(Long userId, Long habitId, ZonedDateTime dateTime) {
-        return modelMapper.map(habitStatusRepo.findByUserIdAndHabitIdAndCreateDate(userId, habitId, dateTime)
-                .orElseThrow(() -> new WrongIdException(ErrorMessage.USER_HAS_NO_STATUS_FOR_SUCH_HABIT)),
+    public HabitStatusDto findStatusByHabitAssignId(Long habitAssignId) {
```

Comment:
This method is really huge and needs refatoring. First, please, move if else statements to private methods and than we will se how can we esle refator it.

Reply:
Created new card with issue.